File: lk/lk_test/fib.c
#include <stdio.h>

int fibR(int n)
{
    if (n < 2) return n;
    return (fibR(n-2) + fibR(n-1));
}


int fibI(int n)
{
    int last = 0;
    int cur = 1;
    n = n - 1;
    while(n)
    {
        --n;
        int tmp = cur;
        cur = last + cur;
        last = tmp;
    }
    return cur;
}


int main()
{
    const int N = 43; //Should return 433494437
    printf("fib: %d = %d\n", fibR(N), fibI(N));
    return 0;
}

File: lk/lk_test/fib.lua
function fibR(n)

    if (n < 2) then return n end
    return (fibR(n-2) + fibR(n-1))
end


function fibI(n)

    local last = 0
    local cur = 1
    n = n - 1
    while (n > 0)
    do
        n = n - 1
        local tmp = cur
        cur = last + cur
        last = tmp
    end
    return cur
end




N = 43 --Should return 433494437
print("fib: " .. fibR(N) .. " = " .. fibI(N))
